src/util.rs

```diff
@@ -485,3 +486,10 @@ where
     let de = &mut serde_json::Deserializer::from_str(v);
     serde_path_to_error::deserialize(de)
 }
+
+pub(crate) fn required_attribute<'a>(key: &'static str, attr: &'a Attributes) -> Result<&'a str, Error> {
```

Comment:
[minor nit] Good call on moving this to here, I would just swap the order of the arguments because `required_attribute(&attr, "x-amz-matdesc")` looks more like `attr["x-amz-matdesc"]` and thus more idiomatic to me

Reply:
Done 👍